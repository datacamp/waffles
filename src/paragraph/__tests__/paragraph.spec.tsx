import { render } from '@testing-library/react';

import { Paragraph } from '../index';

const variants = ['primary', 'secondary'] as const;
const sizes = ['small', 'medium', 'large'] as const;

describe('Paragraph', () => {
  it('renders a paragraph containing the text', () => {
    const { container } = render(
      <Paragraph>Short paragraph about Taylor Swift.</Paragraph>,
    );

    const paragraph = container.querySelector('p');

    expect(paragraph).toBeInTheDocument();
    expect(paragraph).toHaveTextContent('Short paragraph about Taylor Swift.');
  });

  it('sets the data attribute on the paragraph', () => {
    const { getByTestId } = render(
      <Paragraph data-testid="paragraph-22">
        Short paragraph about Taylor Swift.
      </Paragraph>,
    );

    const paragraph = getByTestId('paragraph-22');

    expect(paragraph).toBeInTheDocument();
  });

  describe('renders snapshot of', () => {
    variants.forEach((variant) => {
      sizes.forEach((size) => {
        it(`variant ${variant} and size ${size}`, () => {
          const { container } = render(
            <Paragraph variant={variant} size={size}>
              Test
            </Paragraph>,
          );

          const paragraph = container.querySelector('p');
          expect(paragraph).toMatchSnapshot();
        });
      });
    });
  });

  describe('renders snapshot of inverted', () => {
    variants.forEach((variant) => {
      it(`variant ${variant}`, () => {
        const { container } = render(
          <Paragraph variant={variant} inverted>
            Test
          </Paragraph>,
        );

        const paragraph = container.querySelector('p');
        expect(paragraph).toMatchSnapshot();
      });
    });
  });
});
