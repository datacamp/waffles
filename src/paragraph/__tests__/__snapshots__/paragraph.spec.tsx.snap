// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph renders snapshot of inverted variant primary 1`] = `
.emotion-0 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  line-height: 1.5;
  margin-bottom: 8px;
  color: #FFFFFF;
}

<p
  class="emotion-0"
>
  Test
</p>
`;

exports[`Paragraph renders snapshot of inverted variant secondary 1`] = `
.emotion-0 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  line-height: 1.5;
  margin-bottom: 8px;
  color: #9BA3AB;
}

<p
  class="emotion-0"
>
  Test
</p>
`;

exports[`Paragraph renders snapshot of variant primary 1`] = `
.emotion-0 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  line-height: 1.5;
  margin-bottom: 8px;
  color: #05192D;
}

<p
  class="emotion-0"
>
  Test
</p>
`;

exports[`Paragraph renders snapshot of variant secondary 1`] = `
.emotion-0 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  line-height: 1.5;
  margin-bottom: 8px;
  color: #65707C;
}

<p
  class="emotion-0"
>
  Test
</p>
`;
